test/types/config_types.ts

```diff
@@ -63,3 +63,40 @@ chart4.data.datasets.push({
   type: 'line',
   data: [1, 2, 3]
 });
+
+const chart5 = new Chart('chart', {
+  data: {
+    labels: ['1', '2', '3'],
+    datasets: [{
+      type: 'line',
+      data: [{
+        category: '1',
+        value: 1
+      }]
+    }]
+  }
+});
+
+chart5.data.datasets.push({
+  type: 'line',
+  data: [{
+    category: '1',
+    value: 1
+  }]
+});
+
+chart5.config.data.datasets.push({
+  type: 'line',
+  data: [{
+    category: '2',
+    value: 2
+  }]
+});
+
+// number | Point | [number, number] | BubbleDataPoint | { category: string, value: number }
```

Comment:
This is incorrect, this should not return the bubble datapoint

Reply:
@LeeLenaleee 
Is correct because:

```ts
// type T = number | [number, number] | Point | BubbleDataPoint
type T = ChartData['datasets'][number]['data'][number];
```

Plus, in a runtime data can be changed in any way

Reply:
@LeeLenaleee anyway, it is a problem with `ChartData`, not with changes from #10963, and it should be fixed in a separate PR.

Reply:
ChartData was changed in that pr, so I'm not sure how its not related. Afaik we can not achieve same level of detail in the types without templating.

Reply:
This did work before #10963 so it should either be fixed in this PR or #10963 should be totally reverted.

As you can see in this [codesandbox](https://codesandbox.io/s/react-chartjs-2-chart-js-issue-template-v4-forked-n31b0o?file=/src/App.tsx)

On line 31 I try to read the r prop and it does not let me because it does not exist

Reply:
@kurkle 

> ChartData was changed in that pr

`ChartData` interface and other interfaces that are used in `ChartData`  were not changed in that PR

@LeeLenaleee 

`ChartData` is not `ChartData<"line", number[], string>`. It worked because the data property in that chart.js version is locking for input data types, so in that case datasets with other chart type can't be added, and all tests from this file are failing: https://github.com/TrigenSoftware/Chart.js/blob/97772186316a70add5d5b262489b6409913bf4e3/types/tests/config_types.ts

Reply:
In my oppinion that would be better for now because having the current that you can access bubble datapoint on a line dataset by default is worse in my oppinion